src/lib/arch/riscv/Architecture.cc

```diff
@@ -70,24 +81,28 @@ Architecture::Architecture(kernel::Linux& kernel, YAML::Node config)
       }
     }
     // Store any opcode-based latency override
-    for (size_t j = 0; j < port_node["Instruction-Opcode"].size(); j++) {
-      uint16_t opcode = port_node["Instruction-Opcode"][j].as<uint16_t>();
+    for (size_t j = 0; j < port_node["Instruction-Opcodes"].num_children();
+         j++) {
+      uint16_t opcode;
+      port_node["Instruction-Opcodes"][j] >> opcode;
       opcodeExecutionInfo_[opcode].latency = latency;
       opcodeExecutionInfo_[opcode].stallCycles = throughput;
     }
   }
 
   // ports entries in the groupExecutionInfo_ entries only apply for models
   // using the outoforder core archetype
-  if (config["Core"]["Simulation-Mode"].as<std::string>() == "outoforder") {
+  if (config::SimInfo::getSimMode() == config::SimulationMode::Outoforder) {
     // Create mapping between instructions groups and the ports that support
     // them
-    for (size_t i = 0; i < config["Ports"].size(); i++) {
+    for (size_t i = 0; i < config["Ports"].num_children(); i++) {
       // Store which ports support which groups
-      YAML::Node group_node = config["Ports"][i]["Instruction-Group-Support"];
-      for (size_t j = 0; j < group_node.size(); j++) {
-        uint16_t group = group_node[j].as<uint16_t>();
-        uint8_t newPort = static_cast<uint8_t>(i);
+      ryml::ConstNodeRef group_node =
+          config["Ports"][i]["Instruction-Group-Support-Nums"];
+      for (size_t j = 0; j < group_node.num_children(); j++) {
+        uint16_t group;
+        group_node[j] >> group;
+        uint16_t newPort = static_cast<uint16_t>(i);
```

Comment:
Does this need to be uint16_t rather than uint8_t?

Reply:
Was changed as we've used more than 256 ports in simulations prior

Reply:
I think at some point it might be nice to use something like:
```cpp
template <typename I, typename J> static std::enable_if_t<std::is_signed_v<I> && std::is_signed_v<J>, I> int_cast(J value) {
  if (value < std::numeric_limits<I>::min() || value > std::numeric_limits<I>::max()) {
    throw std::out_of_range("out of range");
  }
  return static_cast<I>(value);
}

template <typename I, typename J> static std::enable_if_t<std::is_signed_v<I> && std::is_unsigned_v<J>, I> int_cast(J value) {
  if (value > static_cast<std::make_unsigned_t<I>>(std::numeric_limits<I>::max())) {
    throw std::out_of_range("out of range");
  }
  return static_cast<I>(value);
}

template <typename I, typename J> static std::enable_if_t<std::is_unsigned_v<I> && std::is_signed_v<J>, I> int_cast(J value) {
  if (value < 0 || static_cast<std::make_unsigned_t<J>>(value) > std::numeric_limits<I>::max()) {
    throw std::out_of_range("out of range");
  }
  return static_cast<I>(value);
}

template <typename I, typename J> static std::enable_if_t<std::is_unsigned_v<I> && std::is_unsigned_v<J>, I> int_cast(J value) {
  if (value > std::numeric_limits<I>::max()) {
    throw std::out_of_range("out of range");
  }
  return static_cast<I>(value);
}
```

And then we can do:
```
uint16_t newPort = int_cast<uint16_t>(i);
```
So that it doesn't blow up if it does go out of bounds.

Reply:
I think this is starting to go against our "simple cpp" matra